### DISCUSSION TITLE
Table interpolation using 'time elapsed' (or 'shifted time') value

### DISCUSSION THREAD

**Discussion by caolwec** [idaholab/moose#29160] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I hope to obtain 'y' values interpolated from a (x, y) table (read in at initial start), where x is the time elapsed from a starting time determined during the calculation. 'x' can be calculated using a postprocessor, but is there a way to use it to get interpolated 'y' values from the (x, y) table.

**Comment by GiudGiud** [comment]
Hello

I would use the elapsed time as a functor.
Then pass it through an object that will use the functor in lieu of time.
https://mooseframework.inl.gov/source/auxkernels/FunctorCoordinatesFunctionAux.html

now you need a Function that does this. I should have done this instead. I ll make a PR


Guillaume

**Reply by GiudGiud** [reply]
what is the end goal here? what type of Function and in which object is it being fed? 
To make sure that a new Function is actually appropriate

**Reply by caolwec** [reply]
> Hello
> 
> I would use the elapsed time as a functor. Then pass it through an object that will use the functor in lieu of time. https://mooseframework.inl.gov/source/auxkernels/FunctorCoordinatesFunctionAux.html
> 
> now you need a Function that does this. I should have done this instead. I ll make a PR
> 
> Guillaume

Thanks! This works through an Aux variable. You are right - what I need exactly could be just a similar type of 'Function'.

**Comment by caolwec** [comment]
Thanks for the response! My table of (x, y) data is read in through a 'PiecewiseLinear' function, and the interpolated function value will be used in a 'FunctionAux'.

**Reply by GiudGiud** [reply]
this already does what you need then.
https://mooseframework.inl.gov/source/auxkernels/FunctorCoordinatesFunctionAux.html

You can use it instead of the FunctionAux
with a postprcoessor as the t_functor, which takes care of doing the time offset, and which you define using a ParsedPostprocessor (expression = t - time_offset)
